src/components/postcard/PostCard.js

```diff
@@ -0,0 +1,121 @@
+import { React, useState } from "react";
+import { useDispatch } from "react-redux";
+import { isPostLiked } from "../../features";
+import { useAuth } from "../../features/auth/authSlice";
+import {
+  addLikeToPost,
+  removeLikedPost,
+  usePosts,
+  setEditPostMode,
+  deletePost,
+} from "../../features/posts/postSlice";
+import { AddPostModal } from "../../modals";
+import { useNavigate } from "react-router-dom";
+
+export function PostCard({ post }) {
+  const navigate = useNavigate();
+  const [showModal, setShowModal] = useState(false);
+  const { isPostInEditMode } = usePosts();
+  const { user } = useAuth();
+  const dispatch = useDispatch();
+
+  const isPostAlreadyLiked = isPostLiked(user, post.likes);
+
+  const likePost = (id) => {
+    if (!isPostAlreadyLiked) {
+      dispatch(addLikeToPost(id));
+    } else {
+      dispatch(removeLikedPost(id));
+      setChangeColor(false);
+    }
+  };
+  const handleEditPost = (post) => {
+    setShowModal(true);
+    dispatch(setEditPostMode(true));
+  };
+  const handleDeletePost = (postId) => {
+    dispatch(deletePost(postId));
+  };
+  return (
+    <div className="post">
+      {showModal && (
+        <AddPostModal
+          showModal={showModal}
+          setShowModal={setShowModal}
+          isPostInEditMode={isPostInEditMode}
+          postId={post._id}
+        />
+      )}
+      <div className="post-user-detail">
+        <img
+          className="avatar"
+          src="https://i.pravatar.cc/30"
+          alt="profile avatar"
+        />
+        <div className="user-name-time">
+          <div className="profilename">
+            {post.firstName} {post.lastName}
+          </div>
+          <div className="username-date">
+            <div className="username">@{post.username}</div>
+            <div className="post-date">• {post.createdAt}</div>
+          </div>
+        </div>
+      </div>
+      <div className="post-text">{post.content}</div>
+      <div className="like-comment-bookmark-section">
+        <div className="like-comment">
+          <div className="like-count-section">
+            <button
+              className={
+                isPostAlreadyLiked
+                  ? "add-post-footer-btn-blue"
+                  : "add-post-footer-btn"
+              }
+              onClick={() => likePost(post._id)}
+            >
+              {isPostAlreadyLiked ? (
+                <span className="material-symbols-sharp">favorite</span>
+              ) : (
+                <span className="material-symbols-outlined">favorite</span>
+              )}
+            </button>
+            <span>{post.likes?.likeCount} </span>
+          </div>
+          <button
+            className="add-post-footer-btn"
+            onClick={() => {
+              navigate(`/post/${post._id}`);
+            }}
+          >
+            <span class="material-symbols-outlined">comment</span>
+          </button>
```

Comment:
since on click you are only navigating then you can use `Link` over here

Reply:
Done